flocker/ca/_ca.py

```diff
@@ -543,8 +544,11 @@ def initialize(cls, path, authority, hostname, begin=None):
             authority.credential.certificate.original.get_subject(), request,
             serial, EXPIRY_20_YEARS, 'sha256', start=begin,
             additional_extensions=[
-                crypto.X509Extension(
-                    b"subjectAltName", False, alt_name)
+                # subjectAltName overrides common name for validation
+                # purposes, and we want to be able to validate against
+                # "control-service", so we include it too.
+                crypto.X509Extension(b"subjectAltName", False,
+                                     b"DNS:control-service," + alt_name),
```

Comment:
:question: Maybe link here to a follow up issue to add support for multiple host names and IP addresses to the certificate. It seems pretty likely that people will want to connect to the REST API using both DNS names and IP addresses for testing and probably both private and public IP addresses and eventually IPv4 + IPv6 addresses.